### PR TITLE
Pkg.suitesparse merge pkg.template

### PR DESCRIPTION
**Pull request by NeroBurner**
This PR updates the toolchains to the current pkg.template branch and enables some failing tests. Those failing tests are disabled in the branch [test.suitesparse](https://github.com/NeroBurner/hunter/tree/test.suitesparse) with the following two commits
- https://github.com/NeroBurner/hunter/commit/047d12faf8b64671ce15589432977b7164856046
- https://github.com/NeroBurner/hunter/commit/e459579f95cc010f1a3ef024c8e82a4a30d9028b

shall I add those commits to this PR?

* I have checked that this pull request contains only
  `.travis.yml`/`appveyor.yml` changes. All other changes send
  to https://github.com/ruslo/hunter. **[Yes]**

* I have checked that no toolchains removed from CI configs, they are commented
  out instead so other developers can enable them back easily and to simplify
  merge conflict resolution. **[Yes]**

* I have checked that for every commented out toolchain there is a link to the
  broken CI build page or to the minimum compiler requirements documentation
  so other developers can figure out what was the problem exactly. **[No]**
  OSX toolchains break because of missing Fortran compiler, only one linked

* I have checked that for every enabled toolchain corresponding package passed
  all stages of update cycle: test/merge/upload/release. **[Yes]**

### CODE REVIEW

**Comment by ruslo** [comment]
> shall I add those commits to this PR?

Yes. Disable toolchains, then add new version, see:
* https://docs.hunter.sh/en/latest/creating-new/update.html#remove-toolchains

File: .travis.yml [hunk 1]
```diff
@@ -77,30 +77,41 @@ matrix:
...
     # FIXME: https://travis-ci.org/ingenue/hunter/jobs/251076572
-    # - os: osx
-    #   osx_image: xcode9.3
-    #   env: >
-    #     TOOLCHAIN=ios-nocodesign-11-3-dep-9-3
-    #     PROJECT_DIR=examples/SuiteSparse
+    - os: osx
+      osx_image: xcode9.3
+      env: >
+        TOOLCHAIN=ios-nocodesign-11-3-dep-9-3
```
Comment (ruslo) [inline .travis.yml]: Please use `ios-nocodesign-11-4-dep-9-3`:
* https://github.com/ingenue/hunter/blob/ac176586f038f37e7d5909b8ad079ed5aae28fd0/.travis.yml#L87-L91

File: .travis.yml [hunk 2]
```diff
@@ -77,30 +77,41 @@ matrix:
     # OSX {
 
     # FIXME:  https://travis-ci.org/ingenue/hunter/jobs/410420098
-    # - os: osx
-    #   osx_image: xcode9.3
-    #   env: >
-    #     TOOLCHAIN=osx-10-13-make-cxx14
-    #     PROJECT_DIR=examples/SuiteSparse
+    - os: osx
+      osx_image: xcode9.3
```
Comment (ruslo) [inline .travis.yml]: Please use `osx_image: xcode 9.4`:
* https://github.com/ingenue/hunter/blob/ac176586f038f37e7d5909b8ad079ed5aae28fd0/.travis.yml#L75-L79

**Review by ruslo — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 3]
```diff
@@ -29,33 +29,34 @@ dist:
...
 matrix:
   include:
     # Linux {
 
     # FIXME: https://travis-ci.org/ingenue/hunter/jobs/251076560
-    # - os: linux
-    #   env: >
-    #     TOOLCHAIN=clang-cxx17
-    #     PROJECT_DIR=examples/SuiteSparse
+    - os: linux
```
Comment (ruslo) [inline .travis.yml:38]: If this toolchain is working now, then `FIXME` can be removed.

File: .travis.yml [hunk 4]
```diff
@@ -29,33 +29,34 @@ dist:
...
       env: >
         TOOLCHAIN=gcc-7-cxx17
         PROJECT_DIR=examples/SuiteSparse
 
     # FIXME: https://travis-ci.org/ingenue/hunter/jobs/251076562
-    # - os: linux
-    #   env: >
-    #     TOOLCHAIN=android-ndk-r17-api-24-arm64-v8a-clang-libcxx14
-    #     PROJECT_DIR=examples/SuiteSparse
+    - os: linux
```
Comment (ruslo) [inline .travis.yml:48]: Remove `FIXME`?

File: .travis.yml [hunk 5]
```diff
@@ -77,30 +78,44 @@ matrix:
...
-    # - os: osx
-    #   osx_image: xcode9.3
-    #   env: >
-    #     TOOLCHAIN=ios-nocodesign-11-3-dep-9-3
-    #     PROJECT_DIR=examples/SuiteSparse
+    # no fortran compiler
+    #- os: osx
+    #  osx_image: xcode9.4
+    #  env: >
+    #    TOOLCHAIN=ios-nocodesign-11-3-dep-9-3
```
Comment (ruslo) [inline .travis.yml]: `TOOLCHAIN=ios-nocodesign-11-4-dep-9-3`

**Review by ruslo — CHANGES_REQUESTED** [review]

**Comment by NeroBurner** [comment]
thanks for the fast response, removed old FIXME, updated xcode toolchain

edit: forgot to save editor, amended unsaved changes to previous commit

**Review by ruslo — APPROVED** [review]

**Pull request merged**